packages/@aws-cdk/pipelines/lib/actions/publish-assets-action.ts

```diff
@@ -82,6 +82,11 @@ export interface PublishAssetsActionProps {
    * @default - All private subnets.
    */
   readonly subnetSelection?: ec2.SubnetSelection;
+
+  /**
+   * Role arns that might need to be assumed by the `cdk-assets` call
+   */
+  readonly roleArnsForCdkAssets: string;
```

Comment:
This should be called something like `publishingRoleArn`, and again should come from the cloud assembly's asset manifest.